src/main/java/se/iths/springbootgroupproject/controllers/WebController.java

```diff
@@ -108,17 +111,47 @@ public String editUserProfile(@Valid @ModelAttribute("formData") EditUserFormDat
         return "redirect:/web/myprofile";
     }
 
+
+    @GetMapping("/myprofile/editmessage")
+    public String editMessage(Model model, @RequestParam("id") Long id) {
+        Message message = messageService.findById(id);
+        model.addAttribute("formData", new CreateMessageFormData(
+                message.getTitle(),
+                message.getMessageBody(),
+                message.isPrivateMessage()));
+        model.addAttribute("messageId", message.getId());
+        return "editmessage";
+    }
+
+    @PostMapping("/myprofile/editmessage")
+    public String editMessage(@Valid @ModelAttribute("formData") CreateMessageFormData messageForm,
+                              BindingResult bindingResult,
+                              @RequestParam("id") Long id,
+                              RedirectAttributes redirectAttributes) {
+        if (bindingResult.hasErrors()) {
+            redirectAttributes.addAttribute("id",id);
+            return "redirect:/web/myprofile/editmessage";
+        }
```

Comment:
This does redirect it to the correct page, but the errormessages does not work mostly because they are omitted from the HTML-file. I tried adding it to the HTML-file, and I see why have omitted them. With this solution they do not work but we can edit the message, and with just returning the html-page("editmessage") the error messages work but you cannot longer edit the message.
Maybe something to look at later, but right now it works fine. Good job!